aiida/repository/backend/disk_object_store.py

```diff
@@ -118,3 +120,116 @@ def get_object_hash(self, key: str) -> str:
+        override_pack_loose: bool = None,
+        override_do_repack: bool = None,
+        override_clean_storage: bool = None,
+        override_do_vacuum: bool = None,
+    ) -> dict:
+        """Performs maintenance operations.
+
+        :param full:
+            a flag to perform operations that require to stop using the maintained profile.
+        :param override_pack_loose:
+            override flag for forcing the packing of loose files.
+        :param override_do_repack:
+            override flag for forcing the re-packing of already packed files.
+        :param override_clean_storage:
+            override flag for forcing the cleaning of soft-deleted files from the repository.
+        :param override_do_vacuum:
+            override flag for forcing the vacuuming of the internal database when cleaning the repository.
+        :return:
+            a dictionary with information on the operations performed.
+        """
+        from aiida.backends.control import MAINTAIN_LOGGER
+        DOSTORE_LOGGER = MAINTAIN_LOGGER.getChild('disk_object_store')  # pylint: disable=invalid-name
+
+        pack_loose = True
+        do_repack = not live
+        clean_storage = not live
+        do_vacuum = not live
+
+        if live:
+            if override_do_repack or override_clean_storage or override_do_vacuum:
+                errmsg = 'a specifically resquest keyword cannot be applied while the profile is in use:\n'
+                if override_do_repack is not None:
+                    errmsg = ' > override_do_repack = {override_do_repack}\n'
+                if override_clean_storage is not None:
+                    errmsg = ' > override_clean_storage = {override_clean_storage}\n'
+                if override_do_vacuum is not None:
+                    errmsg = ' > override_do_vacuum = {override_do_vacuum}\n'
+                raise ValueError(errmsg)
+
+        if override_pack_loose is not None:
+            pack_loose = override_pack_loose
+
+        if override_do_repack is not None:
+            do_repack = override_do_repack
+
+        if override_clean_storage is not None:
+            clean_storage = override_clean_storage
+
+        if override_do_vacuum is not None:
+            do_vacuum = override_do_vacuum
+
+        if pack_loose:
+            files_numb = self.container.count_objects()['loose']
+            files_size = self.container.get_total_size()['total_size_loose'] * BYTES_TO_MB
+            if dry_run:
+                DOSTORE_LOGGER.report(f'Would pack all loose files ({files_numb} files occupying {files_size} MB) ...')
+            else:
+                DOSTORE_LOGGER.report(f'Packing all loose files ({files_numb} files occupying {files_size} MB) ...')
+                self.container.pack_all_loose()
+
+        if do_repack:
+            files_numb = self.container.count_objects()['packed']
+            files_size = self.container.get_total_size()['total_size_packfiles_on_disk'] * BYTES_TO_MB
```

Comment:
This information is not really adding anything specific to the maintenance operation is it? It just gives the current size, but that doesn't tell what it will be nor what will be saved. Only the latter would be really interesting IMO

Reply:
Well, it can help give you an idea of how long it might take to do the repacking. But ok, I can take it out if you prefer.